Tools/ardupilotwaf/boards.py

```diff
@@ -1139,8 +1147,8 @@ def configure_env(self, cfg, env):
             '-fno-builtin-puts',
             '-mno-thumb-interwork',
             '-mthumb',
-            '--specs=nano.specs',
-            '--specs=nosys.specs',
+            '--specs=%s' % env.NANO_SPECS,
```

Comment:
this doesn't look right, it will mean using the wrong subdir (for fp variances)
```
/arm-none-eabi/lib/nosys.specs
./arm-none-eabi/lib/thumb/v7+fp/softfp/nosys.specs
./arm-none-eabi/lib/thumb/v7+fp/hard/nosys.specs
./arm-none-eabi/lib/thumb/v7-a/nofp/nosys.specs
./arm-none-eabi/lib/thumb/v7-r+fp.sp/softfp/nosys.specs
./arm-none-eabi/lib/thumb/v7-r+fp.sp/hard/nosys.specs
./arm-none-eabi/lib/thumb/v8.1-m.main+mve/hard/nosys.specs
./arm-none-eabi/lib/thumb/v7e-m/nofp/nosys.specs
./arm-none-eabi/lib/thumb/v7/nofp/nosys.specs
./arm-none-eabi/lib/thumb/v7e-m+fp/softfp/nosys.specs
./arm-none-eabi/lib/thumb/v7e-m+fp/hard/nosys.specs
./arm-none-eabi/lib/thumb/v8-m.main+dp/softfp/nosys.specs
./arm-none-eabi/lib/thumb/v8-m.main+dp/hard/nosys.specs
./arm-none-eabi/lib/thumb/v7ve+simd/softfp/nosys.specs
./arm-none-eabi/lib/thumb/v7ve+simd/hard/nosys.specs
./arm-none-eabi/lib/thumb/v7e-m+dp/softfp/nosys.specs
./arm-none-eabi/lib/thumb/v7e-m+dp/hard/nosys.specs
./arm-none-eabi/lib/thumb/v8-a/nofp/nosys.specs
./arm-none-eabi/lib/thumb/v8-m.base/nofp/nosys.specs
./arm-none-eabi/lib/thumb/v6-m/nofp/nosys.specs
./arm-none-eabi/lib/thumb/v8-m.main+fp/softfp/nosys.specs
./arm-none-eabi/lib/thumb/v8-m.main+fp/hard/nosys.specs
./arm-none-eabi/lib/thumb/v7-m/nofp/nosys.specs
./arm-none-eabi/lib/thumb/v8-a+simd/softfp/nosys.specs
./arm-none-eabi/lib/thumb/v8-a+simd/hard/nosys.specs
./arm-none-eabi/lib/thumb/v7-a+fp/softfp/nosys.specs
./arm-none-eabi/lib/thumb/v7-a+fp/hard/nosys.specs
./arm-none-eabi/lib/thumb/nofp/nosys.specs
./arm-none-eabi/lib/thumb/v8-m.main/nofp/nosys.specs
./arm-none-eabi/lib/thumb/v7-a+simd/softfp/nosys.specs
./arm-none-eabi/lib/thumb/v7-a+simd/hard/nosys.specs
```